### DISCUSSION TITLE
type tuple error

### DISCUSSION THREAD

**Discussion by Torres91** [Cyfrin/foundry-full-course-cu#2736] [Q&A]
I am working on the lottery/raffle contract but have an error in the DeployRaffle.s.sol contract in the deployContract function, 
here is my code : 
``  function deployContract() public returns (Raffle, HelperConfig) {
        HelperConfig helperConfig = new HelperConfig();
        HelperConfig.NetworkConfig memory config = helperConfig.getConfig();

        if (config.subscriptionId == 0) {
            //Create subscription
            CreateSubscription createSubscription = new CreateSubscription();
            (config.subscriptionId, config.vrfCoordinator) = createSubscription
                .createSubscription(config.vrfCoordinator);
``  
and this is my error 
``Error (7407): Type tuple(uint256,address) is not implicitly convertible to expected type tuple(uint64,address).
  --> script/DeployRaffle.s.sol:21:62:
   |
21 |             (config.subscriptionId, config.vrfCoordinator) = createSubscription
   |                                                              ^ (Relevant source part starts here and spans across multiple lines).
`` 
Following the course, I can't find what the problem is. Could someone please advise where I might have gone wrong? 

<img width="1113" alt="Screenshot 2024-10-24 at 13 54 12" src="https://github.com/user-attachments/assets/f0048aac-4250-49b8-b097-32859437d3fa">

**Comment by EngrPips** [comment]
Hello, where did you get your `subscriptionId` from?

**Reply by Torres91** [reply]
I didn't it's still at 0. Where should I get the SubscriptionId?

**Reply by EngrPips** [reply]
The error is happening because you give your subId type of `uint64`, but the `createSubscription` script is returning a subId that is of type `uint256`

**Reply by Torres91** [reply]
Thanks, I might just restart the project again :/

**Reply by EngrPips** [reply]
Alright, Friend. Please move at a speed that enables you to understand everything that is going on, remember that understanding the lesson >>>> finishing the lesson.
